app/models/cart_item.rb

```diff
@@ -2,9 +2,34 @@ class CartItem < ApplicationRecord
   belongs_to :product
   belongs_to :cart, optional: true
 
+  before_save :set_unit_price
+  before_save :set_total
+
   CARTITEM_ATTRS = %i(quantity product_id).freeze
 
   validates :quantity, presence: true,
     numericality:
     {only_integer: true, greater_than: Settings.cart_item.validation.greater}
+
+  def unit_price
+    if persisted?
+      self[:unit_price]
+    else
+      product.price
+    end
```

Comment:
```suggestion
    persisted? ? self[:unit_price] : product.price
```
Như thế này đc ko em

Reply:
để em thử ạ